### PR TITLE
Fix menu permissions

### PR DESCRIPTION
**Pull request by wychoong**
Currently `MenuItem` with permission is not hidden from sidebar/menu when staff have no permission to access which results to getting 403 on clicking.

This PR looks to fix:
- add `gate()` to menu registration
- hide `Group/Section` label when no Item is accessible
- Setting menu now redirect to first accessible route instead of always `hub.attributes.index`
- fix Staff permission when remove `Settings` perm remove other permissions
    - eg: Enable `Settings` and `Manage Orders`, save, refresh page permissions still correct, Disable `Settings`, save and refresh, `Manage Orders` permission removed as well

*****

- [x] Documentation updates (n/a)
- [x] Automated tests
- [x] Changelog entries (core and hub)

### CODE REVIEW

File: packages/admin/src/Menu/MenuSlot.php [hunk 1]
```diff
@@ -108,17 +122,55 @@ public function getItems(): Collection
...
      */
     public function getGroups()
     {
-        return $this->groups;
+        return $this->groups->filter(function ($group) {
+            return ! $group->gate || Auth::user()->can($group->gate);
+        });
+    }
+
+    public function hasLinks(): bool
```
Comment (alecritson) [inline packages/admin/src/Menu/MenuSlot.php:142]: Do you think we can refactor a bit to reduce this?

**Review by alecritson — COMMENTED** [review]
Thanks @wychoong Will have a pull down and play around this but looks promising!

File: packages/admin/src/Menu/MenuSlot.php (hunk 1 above)
Comment (wychoong) [reply packages/admin/src/Menu/MenuSlot.php:142]: I don't think there is a meaningful way to refactor these as `items`, `groups` and `sections` are stored separately properties

File: packages/admin/src/Menu/MenuSlot.php [hunk 2]
```diff
@@ -108,17 +122,55 @@ public function getItems(): Collection
...
+        }
+
+        if ($this->getSections()->filter(fn ($section) => $section->hasLinks())->count()) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public function getFirstLink()
```
[the remark this answers is not in the document]
Comment (wychoong) [reply packages/admin/src/Menu/MenuSlot.php:159]: same

File: packages/admin/src/Menu/MenuSlot.php (hunk 1 above)
Comment (wychoong) [reply packages/admin/src/Menu/MenuSlot.php:142]: I'm open to work on if you can give me some pointer to start with

**Pull request closed without merging**